src/Npgsql/NpgsqlCommand.cs

```diff
@@ -311,6 +317,24 @@ private set
             }
         }
 
+        /// <summary>
+        /// Gets or sets whether this query will be persisted when it is prepared. This will keep the
+        /// compiled statement available after the command is disposed or the connection is returned to the pool.
+        /// </summary>
+        [DefaultValue(DefaultIsPersistent)]
+        public bool IsPersistent
+        {
+            get { return _isPersistent ?? (_connection?.PersistPrepared ?? DefaultIsPersistent); }
+            set
+            {
+                _isPersistent = value;
+                if (!value)
+                {
+                    DeallocatePrepared(true);
```

Comment:
Yeah I also thought about this, propably it is better that the statement is deallocated when it is actually disposed.